### PR TITLE
Improve ip manger performance

### PR DESCRIPTION
**Pull request by DavidLiu506**
This is for ip manager performance improvement.

1 Transaction on ipAddrRangeCache (not include vpcIpRangeCache.get(vpcId)), this can reduce distribute lock and improve concurrent cache operation. 

2 Add retry logic code in port manager is for cocurrent operate on create subnet and allocate ip.

### CODE REVIEW

File: services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java [hunk 1]
```diff
@@ -190,17 +196,9 @@ private IpAddrAlloc doAllocateIpAddr(String vpcId, int ipVersion, String ipAddr)
      * @throws Exception Db operation or ip address assignment exception
      */
     @DurationStatistics
-    public synchronized IpAddrAlloc allocateIpAddr(IpAddrRequest request) throws Exception {
-        IpAddrAlloc ipAddrAlloc = null;
-        try (Transaction tx = ipAddrRangeCache.getTransaction().start()) {
-            ipAddrAlloc = allocateIpAddrMethod(request);
-            tx.commit();
+    public IpAddrAlloc allocateIpAddr(IpAddrRequest request) throws Exception {
+            IpAddrAlloc ipAddrAlloc = allocateIpAddrMethod(request);
```
Comment (cj-chung) [inline services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java:200]: @DavidLiu506 Do we still need this function if we push all the process code into `allocateIpAddrMethod`?

File: web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java [hunk 2]
```diff
@@ -84,11 +84,18 @@ public IpAddrRequest allocateIpAddress(IpVersion ipVersion, String vpcId, String
...
+        while(true) {
+            try {
+                IpAddrRequest result = restTemplate.postForObject(ipManagerUrl, request, IpAddrRequest.class);
+                verifyAllocatedIpAddr(result);
+                return result;
+            } catch (Exception e) {
+                // handle exception
+                if (++count == maxTries) throw e;
+            }
+        }
```
Comment (cj-chung) [inline web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java:98]: @DavidLiu506 Should we use formal way to do the retry? such as `RetryTemplate` described in this link: https://www.baeldung.com/spring-retry

Comment (DavidLiu506) [reply web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java:98]: Nice. Thank you!

File: services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java (hunk 1 above)
Comment (DavidLiu506) [reply services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java:200]: It called by other functions.

File: web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java (hunk 2 above)
Comment (DavidLiu506) [reply web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java]: @cj-chung fixed using spring framework.

**Review by cj-chung — APPROVED** [review]
LGTM

**Comment by xieus** [comment]
@DavidLiu506 Please add a brief description on the PR as well.

**Pull request merged**